apps/web/src/lib/services/job.service.ts

```diff
@@ -1043,25 +1045,34 @@ export const jobService = (() => {
           }
 
           // If the latest job event is the same as the agent job status result, return the current job status
-          if (latestJobEvent.externalId === agentJobStatusResult.data.id) {
+          if (
+            latestJobEvent.externalId === agentJobStatusResult.data.status_id
+          ) {
             return computeJobStatus(job);
           } else {
             // If the agent job status result has no external ID, check if the latest job event status is the same as the agent job status
-            if (!agentJobStatusResult.data.id) {
+            if (!agentJobStatusResult.data.status_id) {
               if (latestJobEvent.status === agentJobStatus) {
                 return computeJobStatus(job);
               }
             }
           }
 
+          const inputSchemaData = agentJobStatusResult.data.input_schema;
+          const inputSchemaValue = inputSchemaData
+            ? "input_data" in inputSchemaData
+              ? inputSchemaData.input_data
+              : inputSchemaData.input_groups
+            : null;
+
           const newJobEvent = await jobEventRepository.createJobEventForJobId(
             job.id,
             {
-              externalId: agentJobStatusResult.data.id,
+              externalId: agentJobStatusResult.data.status_id,
               status: agentJobStatus,
-              inputSchema: JSON.stringify(
-                agentJobStatusResult.data.input_schema,
-              ),
+              inputSchema: inputSchemaValue
+                ? JSON.stringify(inputSchemaValue)
+                : null,
```

Comment:
Should we do this here or in the repo layer? I would prefer this in the repo layer tbh. What if a caller forgets it?